hatchet_sdk/clients/events.py

```diff
@@ -72,40 +77,45 @@ async def async_bulk_push(
 
     @tenacity_retry
     def push(self, event_key, payload, options: PushEventOptions = None) -> Event:
-        namespace = self.namespace
+        with self.otel_tracer.start_as_current_span("hatchet.run") as span:
+            carrier = {}
+            TraceContextTextMapPropagator().inject(carrier)
 
-        if (
-            options is not None
-            and "namespace" in options
-            and options["namespace"] is not None
-        ):
-            namespace = options["namespace"]
-            del options["namespace"]
+            namespace = self.namespace
 
-        namespaced_event_key = namespace + event_key
+            if (
+                options is not None
+                and "namespace" in options
+                and options["namespace"] is not None
+            ):
+                namespace = options["namespace"]
+                del options["namespace"]
 
-        try:
-            meta = None if options is None else options["additional_metadata"]
-            meta_bytes = None if meta is None else json.dumps(meta).encode("utf-8")
-        except Exception as e:
-            raise ValueError(f"Error encoding meta: {e}")
+            namespaced_event_key = namespace + event_key
 
-        try:
-            payload_bytes = json.dumps(payload).encode("utf-8")
-        except json.UnicodeEncodeError as e:
-            raise ValueError(f"Error encoding payload: {e}")
-
-        request = PushEventRequest(
-            key=namespaced_event_key,
-            payload=payload_bytes,
-            eventTimestamp=proto_timestamp_now(),
-            additionalMetadata=meta_bytes,
-        )
+            try:
+                meta = dict() if options is None else options["additional_metadata"]
```

Comment:
changed the default here from `None` to `dict` so even if `options` is `None`, we can still pass the trace context through